File: Tarea11 (RadixSort)/Tarea11.cs
using System;

class Tarea11 {
    static void CountingSort(int[] arr, int exp) {
        int s = arr.Length;
        int[] outputArray = new int[s];
        int[] countArray = new int[10];

        for (int j = 0; j < s; j++) {
            int idx = (arr[j] / exp) % 10;
            countArray[idx]++;
        }

        for (int j = 1; j < 10; j++)
            countArray[j] += countArray[j - 1];

        for (int j = s - 1; j >= 0; j--) {
            int idx = (arr[j] / exp) % 10;
            outputArray[countArray[idx] - 1] = arr[j];
            countArray[idx]--;
        }

        for (int j = 0; j < s; j++)
            arr[j] = outputArray[j];
    }

    static void RadixSort(int[] arr) {
        int max1 = arr[0];
        foreach (int num in arr)
            if (num > max1) max1 = num;

        for (int exp = 1; max1 / exp > 0; exp *= 10)
            CountingSort(arr, exp);
    }

    static void Main() {
        int[] arr = {171, 46, 76, 91, 803, 25, 3, 67};
        Console.WriteLine("Arreglo antes de ordenar:");
        Console.WriteLine(string.Join(" ", arr));

        RadixSort(arr);

        Console.WriteLine("Después de ordenar el arreglo:");
        Console.WriteLine(string.Join(" ", arr));
    }
}

File: Tarea11 (RadixSort)/Tarea11.js
function countingSort(arr, exp) {
    let s = arr.length;
    let outputArray = new Array(s).fill(0);
    let countArray = new Array(10).fill(0);

    for (let j = 0; j < s; j++) {
        let idx = Math.floor(arr[j] / exp) % 10;
        countArray[idx]++;
    }

    for (let j = 1; j < 10; j++)
        countArray[j] += countArray[j - 1];

    for (let j = s - 1; j >= 0; j--) {
        let idx = Math.floor(arr[j] / exp) % 10;
        outputArray[countArray[idx] - 1] = arr[j];
        countArray[idx]--;
    }

    for (let j = 0; j < s; j++)
        arr[j] = outputArray[j];
}

function radixSort(arr) {
    let max1 = Math.max(...arr);
    for (let exp = 1; Math.floor(max1 / exp) > 0; exp *= 10)
        countingSort(arr, exp);
}

let arr = [171, 46, 76, 91, 803, 25, 3, 67];
console.log("Arreglo antes de ordenar:");
console.log(arr.join(" "));

radixSort(arr);

console.log("Después de ordenar el arreglo:");
console.log(arr.join(" "));
